### PR TITLE
all challenges complete

### PR DESCRIPTION
**Pull request by danielforkner**

### CODE REVIEW

**Comment by TheCSharpAcademy** [comment]
@danielforkner hi Daniel, I just wanted to double check if you are a member in The C# Academy website, as I couldn't find your details.

**Comment by danielforkner** [comment]
@TheCSharpAcademy I am now! I also submitted the PR link in the project page. Thank you!

File: danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
+﻿using Newtonsoft.Json;
+using System.Linq;
```
Comment (TheCSharpAcademy) [inline danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs:2]: Please remove unused usings

File: danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+namespace CalculatorLibrary
```
Comment (TheCSharpAcademy) [inline danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs:4]: not mandatory but you can use the newest namespaces to better utilize space: 
image

File: danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+{
+    public class Calculation
+    {
+        public int Id { get; set; }
+        public double Operand1 { get; set; }
+        public double Operand2 { get; set; }
+        public string Operation { get; set; }
+        public double Result { get; set; }
+        public bool Success { get; set; }
+    }
```
Comment (TheCSharpAcademy) [inline danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs:14]: Add an empty line between methods and between classes

File: danielforkner.Calculator/danielforkner.Calculator/Program.cs [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+            int idx = 0;
+            while (viewingOperations)
+                {
+                    if (calculator.Operations.Count == 0)
+                    {
+                        Console.WriteLine("You don't have any past operations to view!");
+                        viewingOperations = false;
+                        continue;
+                    }
+                    // operation
```
Comment (TheCSharpAcademy) [inline danielforkner.Calculator/danielforkner.Calculator/Program.cs:22]: Maybe these comments came from Microsoft Docs, but usually refrain from commenting what can be explained by the code itself. Code should be self explanatory. The better you name your members, arguments, classes, the less comments you need.

File: danielforkner.Calculator/danielforkner.Calculator/Program.cs [hunk 5]
```diff
@@ -0,0 +1,216 @@
+﻿using CalculatorLibrary;
+
+namespace CalculatorProgram
+{
+
```
Comment (TheCSharpAcademy) [inline danielforkner.Calculator/danielforkner.Calculator/Program.cs:5]: Remove unnecessary empty lines 5 and 8.

**Review by TheCSharpAcademy — CHANGES_REQUESTED** [review]
@danielforkner, everything is looking great, all comments relate to code organization/tidiness. As for the comments, I know Microsoft Docs can be full o them, but it's something you want to think about from now on.

**Comment by TheCSharpAcademy** [comment]
@danielforkner Hey there! Just checking on your progress with this PR? 😁 Do you need any help?

File: danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs (hunk 1 above)
Comment (danielforkner) [reply danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs:2]: Thank you, removed System.Linq

File: danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs (hunk 2 above)
Comment (danielforkner) [reply danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs:4]: Thank you, I will do that!

File: danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs (hunk 3 above)
Comment (danielforkner) [reply danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs:14]: Does this apply to fields and properties as well?

File: danielforkner.Calculator/danielforkner.Calculator/Program.cs (hunk 4 above)
Comment (danielforkner) [reply danielforkner.Calculator/danielforkner.Calculator/Program.cs:22]: Good point, removed!

File: danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs (hunk 3 above)
Comment (TheCSharpAcademy) [reply danielforkner.Calculator/CalculatorLibrary/CalculatorLibrary.cs:34]: @danielforkner no, just methods and classes. It's also good to have some separation between related blocks of code within a method so it doesn't become spaghetti 😁

**Pull request merged**